change/@ni-nimble-components-878686fc-b8ba-42b4-a8d3-ac0bf0919b9d.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
we have one more listbox implementation that is using a different size metric currently. I suspect we want all our dropdowns to behave the same: https://github.com/ni/nimble/blob/51aef514fe30e2cac694b885f3ebec000aed1a43/packages/nimble-components/src/rich-text/mention-listbox/styles.ts#L23

Reply:
@fredvisser should the rich text dropdown (which also shows a large number of items) behave similarly and increase from 5.5 to 10.5 items shown?
![image](https://github.com/ni/nimble/assets/1588923/a824f6ea-221c-4540-8fda-1da3fe268e52)

Reply:
I wouldn't change the rich text @mention dropdown just to maintain consistency with combobox and select. Those controls need to be general purpose, where the @mention dropdown is more narrowly tailored. Let's wait until we specific feedback for the @mention dropdown before changing it.

Reply:
What's the usage specific reason that a list of users should have a different number of values shown? Don't think I follow. As far as I'm aware showing 5.5 items was arbitrarily defined. Guess I should ask what defines 10.5 as a good number and when to use one or the other?

Reply:
Also don't think the rich text impl is as robust and probably should align on being aware of the viewport size. If we do that, do we keep the work of it being intentionally a different number of items?

Reply:
Max 10.5 rows for the select and combobox was an inference from the [select best](https://www.nngroup.com/articles/listbox-dropdown/) [practices](https://designsystem.digital.gov/components/combo-box/#guidance) that there shouldn't be more than 15 items in a (non-filterable) select. 

Virtually all design systems have a limit, but most don't explain or document the rationale. [IBM Carbon](https://carbondesignsystem.com/components/dropdown/usage/#scrolling-behavior) emphasizes showing the half row, and suggests 6.5 as a starting point. Our rows aren't very tall, so 10.5 is a similar height.

When we fix the rich text @mention dropdown to clip/scroll at the view boundary, it wouldn't be unexpected to maintain 5.5 as the limit. The @mention dropdown is invoked while typing, and is a search-first, keyboard-first workflow, so it's reasonable to make a different choice.

Reply:
For dev actions, I think it's clear that we should make the richtext mention listbox viewport aware like we do for select and combobox. @m-akinc I think we could do that either as part of this PR or to keep the review smaller / isolated we can do it in a follow-up PR (preference since sometime richtext changes end up more complex and would be easier to revert / manage standalone).

The discussion about how many rows to show for the richtext does not have to block the PR. I like the idea that the space of 10.5 of our current short rows is what we allocate. If / when we have double height rows for mention options (show user profile + email, etc) then we would show 5.5 options and fill a similar space.

@fredvisser I went and played with the current rich text mention support and I actually think showing more options might be useful because our filtering algorithm isn't very smart. It looks like we filter by any part of the string and then sort alphabetically by the first and then last name. If we filtered and sorted "good" then the option you want would always be early and I can see an argument for needing to show fewer options because it's probably always the first or second. But right now we filter kinda meh.

Like in the following it's pretty unintuitive that typing `@mar` doesn't show "mark" as like one of the first options. ([teams link to gif to avoid showing names public](https://nio365.sharepoint.com/:i:/s/SystemLinkDesignSystem/EagHOJ2PLxVMp2RavBcTxfMBiPuJXddyUkDuw73-r39_pA?e=w1jC13)). I think having the longer list like shown there helps with the idea that you still need to filter more / there are a lot of matches to deal with.

Reply:
Gonna try and resolve this thread quicker offline with @fredvisser

Reply:
Sounds like we can address the richtext mention listbox separately from this PR.

Reply:
> Sounds like we can address the richtext mention listbox separately from this PR.

@m-akinc can you make sure the work is captured. Maybe add it as a task in #1870 and update the PR description to say this PR partially resolves the issue

Reply:
> Gonna try and resolve this thread quicker offline with @fredvisser

Captured initial guidance on 10.5 vs 5.5 options from discussion here: https://github.com/ni/nimble/issues/2216

So we will keep rich text 5.5 options but it does need to be made viewport aware as discussed.

Reply:
Created new bug to track limiting the height of the richtext mention listbox to the viewport: #2217